python/ml4ir/base/model/scoring/scoring_model.py

```diff
@@ -309,7 +313,15 @@ def __update_metrics(self, inputs, y_true, y_pred):
             mask = None
         except AttributeError:
             mask = None
-        self.compiled_metrics.update_state(tf.cast(y_true, tf.float32), y_pred, mask)
+        y_true = tf.cast(y_true, tf.float32)
+
+        # Compute metrics on primary label
+        segments = inputs.get(self.group_metric_feature)
+        for compiled_metric in self.compiled_metrics._metrics:
+            if isinstance(compiled_metric, SegmentMean):
+                compiled_metric.update_state(y_true, y_pred, segments, mask)
+            else:
+                compiled_metric.update_state(y_true, y_pred, mask)
```

Comment:
This is the main change to scorer. The segment metrics require an additional argument. This is the cleanest way I found to pass it.